### ISSUE TITLE
After I use forward I get nan as output for regression task.

### ISSUE THREAD

**Issue by andamun** [KindXiaoming/pykan#438]
Hello, I use this code to predict my data, with 17 inputs and 1 output, I have normalize data using MinMaxScaler, so the value is between 0-1 and check that my data has no inf value. But after running this code all of prediction using forward, I get all the tensor nan output. Anyone know how to fix this problems? 
`import torch
from kan import KAN
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.metrics import r2_score, mean_absolute_error
from sklearn.model_selection import train_test_split

#import the data

X_train, X_temp, y_train, y_temp = train_test_split(X, y, test_size=0.3, random_state=0)
X_val, X_test, y_val, y_test = train_test_split(X_temp, y_temp, test_size=0.5, random_state=0)

# Convert data to torch tensor
train_input = torch.tensor(X_train.to_numpy(), dtype=torch.float32)
train_label = torch.tensor(y_train.to_numpy()[:, None], dtype=torch.float32)
val_input = torch.tensor(X_val.to_numpy(), dtype=torch.float32)
val_label = torch.tensor(y_val.to_numpy()[:, None], dtype=torch.float32)
test_input = torch.tensor(X_test.to_numpy(), dtype=torch.float32)
test_label = torch.tensor(y_test.to_numpy()[:, None], dtype=torch.float32)

# Create KAN
model = KAN(width=[train_input.shape[1],3,1], grid=5, k=11)

dataset = {
    'train_input': train_input,
    'train_label': train_label,
    'val_input': val_input,
    'val_label': val_label,
    'test_input': test_input,
    'test_label': test_label
}

'''subset=[train_input,train_label,val_input,val_label,test_input,test_label]

# Train KAN
results = model.fit(dataset,opt="LBFGS", steps=50,loss_fn=torch.nn.MSELoss())
print(results)

# Predictions of train val and test datasets

test_preds = model.forward(test_input).detach()
print(test_preds)
print(test_preds.shape)`

This is my run result.
`checkpoint directory created: ./model
saving model version 0.0
| train_loss: nan | test_loss: nan | reg: nan | : 100%|█████████████| 50/50 [00:59<00:00,  1.19s/it]
saving model version 0.1
{'train_loss': [array(0.09629563, dtype=float32), array(0.08319506, dtype=float32), array(0.07543463, dtype=float32), array(0.07111915, dtype=float32), array(0.06755204, 
dtype=float32), array(5.705628, dtype=float32), array(0.9530418, dtype=float32), array(0.56650126, dtype=float32), array(0.18321516, dtype=float32), array(0.17559457, dtype=float32), array(5.302342, dtype=float32), array(0.37209165, dtype=float32), array(0.18791643, dtype=float32), array(0.1386429, dtype=float32), array(0.13849099, dtype=float32), array(20.704945, dtype=float32), array(4.6654925, dtype=float32), array(0.1611432, dtype=float32), array(0.15494776, dtype=float32), array(0.1495405, dtype=float32), array(49.160458, dtype=float32), array(8.646499, dtype=float32), array(1.1910256, dtype=float32), array(0.9224345, dtype=float32), array(0.5285308, dtype=float32), 
array(206.93304, dtype=float32), array(1.1360668, dtype=float32), array(1.1011049, dtype=float32), array(1.0998647, dtype=float32), array(1.0998647, dtype=float32), array(419.061, dtype=float32), array(11.309376, dtype=float32), array(9.072084, dtype=float32), array(9.0720415, dtype=float32), array(9.072068, dtype=float32), array(4.8201576e+12, dtype=float32), array(4.8201576e+12, dtype=float32), array(4.820157e+12, dtype=float32), array(4.820157e+12, dtype=float32), array(4.8201565e+12, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32)], 'test_loss': [array(0.1270209, dtype=float32), array(0.15437205, dtype=float32), array(0.16871421, dtype=float32), array(0.18262088, dtype=float32), array(0.19983354, dtype=float32), array(5600.378, dtype=float32), array(5672.561, dtype=float32), array(5667.9927, dtype=float32), array(5081.8896, dtype=float32), array(5140.631, dtype=float32), array(5.0861964, dtype=float32), array(0.33465755, dtype=float32), array(0.19919494, dtype=float32), array(0.27967653, dtype=float32), array(0.29971698, dtype=float32), array(20.622131, dtype=float32), array(4.823307, dtype=float32), array(0.1958251, dtype=float32), array(0.21578985, dtype=float32), array(0.23664539, dtype=float32), array(84.30502, dtype=float32), array(485.62207, dtype=float32), array(29.014437, dtype=float32), array(53.263874, dtype=float32), array(25.521753, dtype=float32), array(348.89355, dtype=float32), array(8.2052355, dtype=float32), array(28.667255, dtype=float32), array(32.3263, dtype=float32), array(32.3263, dtype=float32), array(434.5692, dtype=float32), array(422.78946, dtype=float32), array(371.68704, dtype=float32), array(371.68704, dtype=float32), array(371.68704, dtype=float32), array(13562971., dtype=float32), array(13562971., dtype=float32), array(13562970., dtype=float32), array(13562971., dtype=float32), array(13562970., dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32), array(nan, dtype=float32)], 'reg': [array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(0., dtype=float32), array(nan, dtype=float32)]}
tensor([[nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan],
        [nan]])`

**Comment by KindXiaoming** [comment]
could you please check is there any nan in your inputs?

**Comment by dakeend** [comment]
Hello, I also encountered this problem. May I ask if you have solved it?
